test/e2e/specs/realrp/realrp.go

```diff
@@ -34,6 +34,26 @@ var _ = Describe("Resource provider e2e tests [Real]", func() {
 		}
 	})
 
+	It("should keep the end user from deleting any azure resources", func() {
+		resourcegroup, err := cli.OSAResourceGroup(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"), os.Getenv("AZURE_REGION"))
+		Expect(err).NotTo(HaveOccurred())
+		By(fmt.Sprintf("OSA resource group is %s", resourcegroup))
+
+		pages, err := cli.Resources.ListByResourceGroup(ctx, resourcegroup, "", "", nil)
+		Expect(err).ToNot(HaveOccurred())
+		// attempt to delete all resources in the resourcegroup
+		for pages.NotDone() {
+			for _, v := range pages.Values() {
+				By(fmt.Sprintf("trying to delete %s/%s", *v.Type, *v.Name))
+				_, err := cli.Resources.DeleteByID(ctx, *v.ID)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).Should(ContainSubstring(`Status=409 Code="ScopeLocked"`))
```

Comment:
I would have liked to see a technique which unwraps the error. @jim-minter @kargakis any ideas? This definitely does the job though

Reply:
Do you have any idea why this is a 409 vs a 403?

Reply:
No idea, maybe because you do have permission to access the resource, just not enough privileges to delete it ("Access denied" vs "Conflict")
Thanks for the pointer @asalkeld

Reply:
the managed app implementation is via holding a write lock on the objects, fwiw